### DISCUSSION TITLE
manager-head.html, preview-head.html and preview-body.html not changing for the sidebar styling?

### DISCUSSION THREAD

**Discussion by TonyHoanTrinh** [storybookjs/storybook#15652] [Help]
Hello,

I am trying to change the sidebar styling.
Am I trying to do a very basic example of changing the background color of the sidebar to yellow.

I looked at the sidebar and also inspected the element and found the class to be ```<div class="css-1wpdhyu">```

![image](https://user-images.githubusercontent.com/26545361/126537686-217b48d8-4ce7-420a-b075-798882089ea4.png)

I can change the background-color directly using console tools

![image](https://user-images.githubusercontent.com/26545361/126538005-3b038578-2c8d-4c7e-9146-6cfdf6e22a42.png)

![image](https://user-images.githubusercontent.com/26545361/126538085-614341e3-0165-4f4f-b5ab-e91b62f666fd.png)

I tried to do change the styling using this in all 3 files: 
- **manager-head.html**
- **preview-head.html**
- **preview-body.html**

```
<style>
  .css-1wpdhyu {
    background-color: yellow;
  }
</style>
```

However when I do this, it doesn't change the sidebar background-color to yellow.

How would I go about fixing this?
Any help would be really appreciated!

Thank you so much!

**Comment by shilman** [comment]
Did you try running with `--no-manager-cache`?

cc @ghengeveld @jonniebigodes

**Reply by TonyHoanTrinh** [reply]
Yes I just tried running it with the script ` storybook_cache: "start-storybook --no-manager-cache" ` and I am still not getting the desired result :(

**Reply by shilman** [reply]
Try increasing the specificity of the selector, e.g. `.os-content > .css-1wpdhyu`?

**Reply by TonyHoanTrinh** [reply]
I'm sorry? I have my code above I'm doing this
```
<style>
  .css-1wpdhyu {
    background-color: yellow;
  }
</style>
```

**Reply by jonniebigodes** [reply]
@TonyHoanTrinh I have taken a look at this and I'm going to detail the steps I took to reproduce:
- Cloned the [Intro to Storybook React template](https://github.com/chromaui/intro-storybook-react-template), this is just as a means of convenience to get something already set up.
- Removed the CSS from `src/index.css` file.
- Created a `manager-head.html` inside the `.storybook` folder with the following inside:
   ```html
       <style>
         .os-content{
           background-color: yellow;
         }
      </style>
   ```
-  Opened up a terminal and ran `yarn storybook` and it yields the following:
![Storybook-custom-css-manager-head](https://user-images.githubusercontent.com/22988955/126807777-6f909f54-4a06-45bd-bfa7-6bd808a876fd.png)
Opening up the dev tools I got the following:
![Storybook-custom-css-manager-head-Devtolls](https://user-images.githubusercontent.com/22988955/126807871-a4283f32-ca87-4a36-9ba4-0a7526629b38.png)

As you may have noticed the issue presents itself, the Storybook UI should be showing a yellow background color, but it's not.
- Stopped Storybook.
- Added a new script to my `package.json` file to start Storybook without caching, more specifically:
  ```json
  {
    "scripts": {
       "storybook": "start-storybook -p 6006 -s public",
       "storybook-no-cache": "start-storybook -p 6006 -s public --no-manager-cache",
       "build-storybook": "build-storybook -s public"
    },
   }
   ```
- Ran the `storybook-no-cache` script and it yielded the following:

![Storybook-custom-manager-with-cache](https://user-images.githubusercontent.com/22988955/126815246-14712657-aa2a-4bae-9602-b0de68842f96.png)

And in the devTools:
![Storybook-custom-manager-with-cache-devtools](https://user-images.githubusercontent.com/22988955/126817411-7c7abfe9-4747-4a4b-9016-76a7508a2dd6.png)

As you can see the CSS styles are applied to the whole UI (Storybook Manager). 

- Adjusted the CSS in `.storybook/manager-head.html` to be more specific to target only the class in question, more specifically `.css-1wpdhyu`, turning this into:
   ```html
       <style>
          .os-content > .css-1wpdhyu {
             background-color: yellow;
          }
   
       </style>
  ```
- Started Storybook with ` yarn storybook-no-cache` and it yielded the following:

![Storybook-no-cache-inner-css](https://user-images.githubusercontent.com/22988955/126817805-1935fd94-4d9c-4b7c-babd-570c4e791962.png)

And checked the dev-tools to be sure:

![Storybook-no-cache-inner-css-devtools](https://user-images.githubusercontent.com/22988955/126817827-d5779de3-2f1a-4910-b1ac-a9f5ada89b68.png)

Feel free to provide feedback. If you want I can hoist the code up to a repo so that you can take a look at it. Just let me know.

Stay safe

**Reply by TonyHoanTrinh** [reply]
Hey, can you hoise the repo so I can take a look at it?
I tried it on my project however it doesn't work sadly :(

I have a common.css file that does basically the same thing as index.css and I'm not sure if that is the thing that is causing the background to not change to yellow.

Would having a index.css file affect the result?

**Reply by TonyHoanTrinh** [reply]
Also I wanted to ask, how would you change the font-size? I'm trying to do that as well and having some issues with it.

**Reply by jonniebigodes** [reply]
@TonyHoanTrinh the repo is hoisted [here](https://github.com/jonniebigodes/storybook-custom-manager-css). 

Go through it at your own pace and let me know. 

Regarding this:
> I have a common.css file that does basically the same thing as index.css and I'm not sure if that is the thing that is causing the background to not change to yellow.

It shouldn't affect it unless you're if you're importing it into the HTML file for some reason. 

Also, regarding this item:
> Also I wanted to ask, how would you change the font-size? I'm trying to do that as well and having some issues with it.

If by font-size you mean the size applied to the individual stories and the grouping text, in the example I've mentioned named as  "Example" then you probably need to hunt down the css classes used and adjust accordingly. Feel free to correct me @ghengeveld .

**Reply by TonyHoanTrinh** [reply]
Hello @jonniebigodes ,

Thank you so much!
It works like a charm.
I was wondering is it possible to do custom fonts as well? I'm trying to do font-face with a custom font I downloaded in my node_modules but it doesnt' seem to work.

Again, thank you so much!

**Reply by ghengeveld** [reply]
For a custom font, either load it from a CDN such as Google Fonts, or put it in your static directory (and don't forget to include the -s flag with build-storybook as well).

You probably figured this out already, but you need to use `manager-head` and pass `--no-manager-cache` while developing. Then it's just normal CSS, so you have to account for specificity (look it up). I recommend avoiding the randomly generated class names because they might change between storybook versions.
